app/controllers/posts_controller.rb

```diff
@@ -13,6 +13,13 @@ def new
     @post = Post.new
   end
 
```

Comment:
- Hey Team, we should also add the `load_and_authorize_resource` which sets a `before_filter` that calls these two methods: `load_resource` and `authorize_resource` as the `load_resource` decides whether it should obtain a new instance of the class, in your case (e.g. `Post.new`) or find the particular instance based on `params[:id]` (e.g. `Post.find(params[:id])).` The instance (or a collection of instances for actions like `index`) is assigned to the corresponding instance variable of your controller action.
After this, the ` authorize_resource` is called immediately, checking all your abilities. Let's update this.